src/Patterns/templates/Scd2Snowflake.twig

```diff
@@ -3,88 +3,125 @@
 {% set inputRandomColumn = attribute(inputPrimaryKey, 0) %}
 {% set snapshotInputJoinConditionSql = macro.generateJoin(inputPrimaryKey, "snapshot", "input") %}
 {% set snapshotPrimaryKeySelect = macro.generateSnapshotPrimaryKey(snapshotPrimaryKeyParts, snapshotPrimaryKeyName) %}
+-- SCD2: This method tracks historical data --
+-- by creating new records for new/modified data in the snapshot table. --
 
---- SCD2: This method tracks historical data by creating new records for new/modified data in the snapshot table --
+{% apply noIndent %}
+    {% if config.useDatetime() %}
+        {% set currentDate = '$CURRENT_TIMESTAMP_TXT' %}
+        {% set infiniteDate = '9999-12-31 00:00:00' | quoteValue %}
```

Comment:
Vsetko som podrobne otestovat, ... fixol posledne veci, SNFK je hotovy. Doplnil som aj komentare do kodu, kedze som konecne pochopil ako to cele funguje a kde boli chyby :)